profiler/profiler.go

```diff
@@ -182,15 +185,80 @@ func newProfiler(opts ...Option) (*profiler, error) {
 		prev: make(map[string]*pprofile.Profile),
 	}
 	p.uploadFunc = p.upload
+	p.telemetry = &telemetry.Client{
+		APIKey:    cfg.apiKey,
+		Namespace: telemetry.NamespaceProfilers,
+		Service:   cfg.service,
+		Env:       cfg.env,
+		Client:    p.cfg.httpClient, // use the profiler's http.Client, gives us UDS if that's used
+	}
+
+	// For the initial release, prefer off-by-default rather than
+	// on-by-default
+	if !internal.BoolEnv("DD_INSTRUMENTATION_TELEMETRY_ENABLED", false) {
+		p.telemetry.Disabled = true
+	}
+
+	// For the URL, uploading through agent goes through
+	//	${AGENT_URL}/telemetry/proxy/api/v2/apmtelemetry
+	// for agentless (which we technically don't support):
+	//	https://instrumentation-telemetry-intake.datadoghq.com/api/v2/apmtelemetry
+	// with an API key
+	//
+	// TODO: move this logic into the telemetry package, make the URL field
+	// unnecessary?
+	if cfg.agentless {
+		p.telemetry.URL = "https://instrumentation-telemetry-intake.datadoghq.com/api/v2/apmtelemetry"
+	} else {
+		// TODO: check agent /info endpoint to see if the agent is
+		// sufficiently recent to support this endpiont? overkill?
+		u, err := url.Parse(cfg.agentURL)
+		if err == nil {
+			u.Path = "/telemetry/proxy/api/v2/apmtelemetry"
+			p.telemetry.URL = u.String()
+		}
+		// ??? what do we even do if the URL isn't valid....
```

Comment:
Should this be handled in this PR? Would a log.Fatal be reasonable here? Is there ever a case where the agent URL could be invalid at this point?